version: "3.8"

services:
  postgis:
    image: kartoza/postgis:14-3.1
    volumes:
      - postgis-data:/var/lib/postgresql
    environment:
      - POSTGRES_DB=${POSTGRES_DB}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASS=${POSTGRES_PASS}
      - ALLOW_IP_RANGE=${ALLOW_IP_RANGE}
      - POSTGRES_MULTIPLE_EXTENSIONS=${POSTGRES_MULTIPLE_EXTENSIONS}
    ports:
      - "25432:5432"
    restart: unless-stopped
    networks:
      - osm_proxy

  martin:
    image: urbica/martin
    environment:
      - DATABASE_URL=postgres://renderer:password@postgis:5432/gis
      - DANGER_ACCEPT_INVALID_CERTS=true
    ports:
      - "3000:3000"
    depends_on:
      - postgis
      - importer
    networks:
      - osm_proxy

  importer:
    build:
      context: .
    volumes:
      - importer-diff:/diff
    depends_on:
      - postgis
    networks:
      - osm_proxy
    environment:
      - NATS_IP=nats

volumes:
  postgis-data:
  importer-diff:

networks:
  osm_proxy:
    name: osm_cqrs_network